pkg/bgpv1/gobgp/server.go

```diff
@@ -342,6 +346,28 @@ func (g *GoBGPServer) RemoveNeighbor(ctx context.Context, n types.NeighborReques
 	return nil
 }
 
+// ResetNeighbor resets BGP peering with the provided neighbor address.
+func (g *GoBGPServer) ResetNeighbor(ctx context.Context, r types.ResetNeighborRequest) error {
+	// for this request we need a peer address without prefix
+	peerAddr := r.PeerAddress
```

Comment:
Should `PeerAddress` be `PeerPrefix` since `ResetNeighborRequest` is storing a prefix? Otherwise, maybe L353-354 should be called when storing values of type `PeerAddress`.

Reply:
Not sure, as out CRD API - `CiliumBGPNeighbor.PeerAddress` is actually a prefix and that is how we pass it to `AddNeighbor` / `UpdateNeighbor`, so I think we should be consistent here.